src/modules/previewpane/MarkdownPreviewHandler/MarkdownPreviewHandlerControl.cs

```diff
@@ -0,0 +1,78 @@
+
+        /// <summary>
+        /// Markdown HTML footer.
+        /// </summary>
+        private readonly string htmlFooter = "</div></body></html>";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MarkdownPreviewHandlerControl"/> class.
+        /// </summary>
+        public MarkdownPreviewHandlerControl()
+        {
+            this.extension = new HTMLParsingExtension();
+            this.pipelineBuilder = new MarkdownPipelineBuilder().UseAdvancedExtensions().UseEmojiAndSmiley();
+            this.pipelineBuilder.Extensions.Add(this.extension);
+        }
+
+        /// <summary>
+        /// Start the preview on the Control.
+        /// </summary>
+        /// <param name="dataSource">Path to the file.</param>
+        public override void DoPreview<T>(T dataSource)
+        {
+            this.InvokeOnControlThread(() =>
+            {
+                StringBuilder sb = new StringBuilder();
+                string filePath = dataSource as string;
+                string fileText = File.ReadAllText(filePath);
+                this.extension.BaseUrl = Path.GetDirectoryName(filePath);
+
+                MarkdownPipeline pipeline = this.pipelineBuilder.Build();
+                string parsedMarkdown = Markdown.ToHtml(fileText, pipeline);
+                sb.AppendFormat("{0}{1}{2}", this.htmlHeader, parsedMarkdown, this.htmlFooter);
+
+                WebBrowser browser = new WebBrowser();
+                browser.DocumentText = sb.ToString();
+                browser.Dock = DockStyle.Fill;
+                browser.IsWebBrowserContextMenuEnabled = false;

```

Comment:
How about Unit Test to check these property are set.

Reply:
I will update this!